File: app/src/main/java/me/gabeg/sicksends/SsSportRepository.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Repository for sport problems.
 */
public class SsSportRepository
	extends SsProblemRepository<SsSportDao, SsSport>
{

	/**
	 */
	public SsSportRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.sportDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/sport/SsSportRepository.kt
package me.gabeg.sicksends.db.sport

import me.gabeg.sicksends.db.generic.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for sport problems.
 */
class SsSportRepository @Inject constructor(dao : SsSportDao)
	: SsGenericProblemRepository<SsSportProblem>(dao)
